src/realm/table.cpp

```diff
@@ -2292,17 +2241,16 @@ Obj Table::create_object_with_primary_key(const Mixed& primary_key, FieldValues&
 
     ObjKey unres_key;
     if (m_tombstones) {
-        // Check for potential tombstone
-        GlobalKey object_id{primary_key};
-        ObjKey object_key = global_to_local_object_id_hashed(object_id);
-
-        ObjKey key = object_key.get_unresolved();
-        if (auto obj = m_tombstones->try_get_obj(key)) {
-            auto existing_pk_value = obj.get_any(primary_key_col);
-
-            // If the primary key is the same, the object should be resurrected below
-            if (existing_pk_value == primary_key) {
-                unres_key = key;
+        if (auto sz = m_tombstones->size()) {
+            // Check for potential tombstone
+            Iterator end(*m_tombstones, sz);
+            for (Iterator it(*m_tombstones, 0); it != end; ++it) {
```

Comment:
isn't this going to be slow? maybe we should run some benchmarks.

Reply:
For sure this will be slower than before. The assumption is that we will have a limited number of tombstones. Alternatively we could implement an index just for tombstones.

Reply:
Does the assumption hold true? At least with some benchmark we'd know what to expect.